### PR TITLE
feat: add ruff pre-commit hook for format and lint

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR adds the ruff pre-commit hook for format and lint.

### CODE REVIEW

**Review by EpsilonPrime — APPROVED** [review]

**Comment by danepitkin** [comment]
Thanks, @sanjibansg ! LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ci:
+  autofix_commit_msg: "style: [pre-commit.ci] autofix"
+  autoupdate_commit_msg: "chore(deps): [pre-commit.ci] autoupdate"
```
Comment (westonpace) [inline .pre-commit-config.yaml:3]: Do you need to link the repo to https://pre-commit.ci/ for these `ci` changes to take effect?

Comment (sanjibansg) [reply .pre-commit-config.yaml:3]: Yes, I believe so. I do not have the privileges to sign-in there for this repository.

**Review by westonpace — APPROVED** [review]
Let's merge then and I'll see if I can get the integration working

**Pull request merged**